spec/post_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  user = User.new(name: 'Amanuel', posts_counter: 0)
+  post = Post.new(author: user, title: nil, comments_counter: nil, likes_counter: nil)
+  before { post.save }
+
+  describe 'validations' do
+    it 'requires a title to be present' do
+      expect(post).not_to be_valid
+    end
+
+    it 'requires the title to be at most 250 characters long' do
+      post.title = 'a' * 251
+      expect(post).not_to be_valid
+    end
+
+    it 'requires the comments_counter to be a non-negative integer' do
+      post.comments_counter = -1
+      expect(post).not_to be_valid
+    end
+
+    it 'requires the likes_counter to be a non-negative integer' do
+      post.likes_counter = -1
+      expect(post).not_to be_valid
+    end
```

Comment:
- [ ] You have done a great job on implementing the tests for validations and methods in the models :blush: but it would be nice if you cover all scenarios for the post because you have tested some of the scenarios where your post is **not_valid** Also, it would be nice if test your post model in that scenario where your **comments_counter** is numeric but **likes_counter** is not numeric :+1:

- [ ] Similar to that It would be very great if you cover that scenario of your post where it is **valid** :thinking:

**Note: You have already done it in the user_spec.rb :+1: The same way you can do it in post_spec.rb file :+1:**


### Additional Information :memo:

**Question** :red_circle:
Why should I cover all the scenarios it is not mentioned in the project requirements?

**Answer** :green_circle:
Partially, you are right that it is not mentioned in the project requirement but the whole purpose of testing is that you have to analyze your application's behavior in all scenarios to see if it is working as expected :+1: